src/Components/Web.Extensions/src/wwwroot/headManager.js

```diff
@@ -0,0 +1,51 @@
+// Local helpers
+
+const getHeadElement = (tagName, id) => {
+    const elements = Array.from(document.getElementsByTagName(tagName));
+    return elements.find(e => e._blazorId === id);
+}
+
+const getOrCreateHeadElement = (tagName, id) => {
+    let element = getHeadElement(tagName, id);
+
+    if (element) {
+        return element;
+    }
+
+    element = document.createElement(tagName);
+    element._blazorId = id;
+
+    const head = document.getElementsByTagName('head')[0];
+    head.appendChild(element);
+
+    return element;
+};
+
+// Exported functions
+
+const setTitle = (title) => {
+    document.title = title;
+};
+
+const setTag = (tagName, id, attributes) => {
+    let tag = getOrCreateHeadElement(tagName, id);
+
+    if (attributes) {
+        for (let key in attributes) {
+            if (attributes.hasOwnProperty(key)) {
+                tag.setAttribute(key, attributes[key]);
+            }
+        }
+    }
+};
+
+const removeTag = (tagName, id) => {
+    let tag = getHeadElement(tagName, id);
+    tag && tag.remove();
+};
+
+window._blazorHeadManager = {
+    setTitle,
+    setTag,
+    removeTag,
+};
```

Comment:
We're hoping to have JS encapsulation in .NET 5 which would make it easy to have a real ES6 module here with regular exports. It's not implemented yet so what you're doing here makes sense.

But in the meantime, would it be possible to tweak this file to avoid adding extra consts into global scope? For example using the [IIFE pattern](https://en.wikipedia.org/wiki/Immediately_invoked_function_expression), like this:

```js
(function() {
  function myInternalFunction() {
    ....
  }

  function myExportedFunction() {
    myInternalFunction();
  }

  window._blazorHeadManager = {
    myExportedFunction,
  };
})();
```